File: CPP/FilmTayang.cpp
#include <string>

#include "FilmBioskop.cpp"

using namespace std;

class FilmTayang : public FilmBioskop {
private:
    int hargaTiket;
    string jamTayang;
    string statusTayang;

public:
    // Constructor default
    FilmTayang() : FilmBioskop() {
        hargaTiket = 0;
        jamTayang = "";
        statusTayang = "";
    }

    // Constructor berparameter
    FilmTayang(
        int id,
        string judul,
        string genre,
        int durasi,
        string studio,
        string klasifikasiUsia,
        int hargaTiket,
        string jamTayang,
        string statusTayang
    ) : FilmBioskop(
        id,
        judul,
        genre,
        durasi,
        studio,
        klasifikasiUsia
    ) {

        this->hargaTiket = hargaTiket;
        this->jamTayang = jamTayang;
        this->statusTayang = statusTayang;
    }

    // Getter
    int getHargaTiket() const {
        return hargaTiket;
    }

    string getJamTayang() const {
        return jamTayang;
    }

    string getStatusTayang() const {
        return statusTayang;
    }

    // Setter
    void setHargaTiket(int hargaTiket) {
        this->hargaTiket = hargaTiket;
    }

    void setJamTayang(string jamTayang) {
        this->jamTayang = jamTayang;
    }

    void setStatusTayang(string statusTayang) {
        this->statusTayang = statusTayang;
    }

    // Destructor
    ~FilmTayang() override {
    }
};

File: Java/FilmTayang.java
public class FilmTayang extends FilmBioskop {
    private int hargaTiket;
    private String jamTayang;
    private String statusTayang;

    // Constructor default
    public FilmTayang() {
        super();
        this.hargaTiket = 0;
        this.jamTayang = "";
        this.statusTayang = "";
    }

    // Constructor berparameter
    public FilmTayang(
        int id,
        String judul,
        String genre,
        int durasi,
        String studio,
        String klasifikasiUsia,
        int hargaTiket,
        String jamTayang,
        String statusTayang
    ) {
        super(id, judul, genre, durasi, studio, klasifikasiUsia);
        this.hargaTiket = hargaTiket;
        this.jamTayang = jamTayang;
        this.statusTayang = statusTayang;
    }

    // Getter
    public int getHargaTiket() { return hargaTiket; }
    public String getJamTayang() { return jamTayang; }
    public String getStatusTayang() { return statusTayang; }

    // Setter
    public void setHargaTiket(int hargaTiket) { this.hargaTiket = hargaTiket; }
    public void setJamTayang(String jamTayang) { this.jamTayang = jamTayang; }
    public void setStatusTayang(String statusTayang) { this.statusTayang = statusTayang; }
}
